uat/custom-components/client-java-paho/src/main/java/com/aws/greengrass/testing/mqtt5/client/grpc/GRPCControlServer.java

```diff
@@ -434,6 +428,49 @@ public void subscribeMqtt(MqttSubscribeRequest request, StreamObserver<MqttSubsc
             responseObserver.onCompleted();
         }
 
+        /**
+         * Handler of UnsubscribeMqtt gRPC call.
+         *
+         * @param request incoming request
+         * @param responseObserver response control
+         */
+        @Override
+        public void unsubscribeMqtt(MqttUnsubscribeRequest request,
+                                    StreamObserver<MqttSubscribeReply> responseObserver) {
+
+            int timeout = request.getTimeout();
+            if (!isValidTimeout(timeout, "unsubscribe", responseObserver)) {
+                return;
+            }
+
+            List<String> filters = request.getFiltersList();
+            if (filters.isEmpty()) {
+                logger.atWarn().log("empty filters list");
+                responseObserver.onError(Status.INVALID_ARGUMENT
+                        .withDescription("empty filters list")
+                        .asRuntimeException());
+                return;
+            }
+
+            int connectionId = request.getConnectionId().getConnectionId();
+            MqttConnection connection = mqttLib.getConnection(connectionId);
+            if (!isValidConnection(connection, connectionId, responseObserver)) {
+                return;
+            }
+
+            logger.atInfo().log("Unsubscribe: connectionId {} for {} filters",
+                    connectionId, filters);
+            List<Mqtt5Properties> userProperties = request.getPropertiesList();
+            MqttSubscribeReply unsubscribeReply = connection.unsubscribe(timeout, filters, userProperties);
```

Comment:
UNSUBACK as well as SUBACK has user properties.
Please check https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901193 

Are you going to add user properties for unsubscribe in other PR ?